CHANGELOG.md

```diff
@@ -1,5 +1,56 @@
+ * Support `files.trimTrailingWhitespace` setting with experimental formatter. See [#800](https://github.com/redhat-developer/vscode-xml/pull/800).
+ * Update `xml.format.joinContentLines` settings documentation. See [#817](https://github.com/redhat-developer/vscode-xml/pull/817).
+ * Remove `xml.format.preserveEmptyContent` setting for experimental formatter. See [lemminx#1346](https://github.com/eclipse/lemminx/issues/1346).
+ * Improve text content formatting for experimental formatter. See [lemminx#1331](https://github.com/eclipse/lemminx/issues/1331).
+
+### Performance
+
+ * Improve formatting performance with `xml.format.grammarAwareFormatting` setting. See [lemminx#1368](https://github.com/eclipse/lemminx/issues/1368).
+
+### Bug Fixes
+
+ * Relax NG schema validation not working when DOCTYPE declaration missing with file association. See [#831](https://github.com/redhat-developer/vscode-xml/issues/831).
+ * XSD based Autocompletion for substitutionGroup. See [#825](https://github.com/redhat-developer/vscode-xml/issues/825).
+ * Applying XML completion generates invalid XML content. See [#815](https://github.com/redhat-developer/vscode-xml/pull/815).
+ * no svg validation if `jock.svg` is also installed. See [#805](https://github.com/redhat-developer/vscode-xml/issues/805).
+ * Referenced RelaxNG schema not found via XML catalog when using `xml-model`. See [#804](https://github.com/redhat-developer/vscode-xml/issues/804).
+ * Generate and bind schema doesn't work with experimental formatter. See [#802](https://github.com/redhat-developer/vscode-xml/issues/802).
+ * Fix unexpected behavior of `xml.format.preservedNewlines` with experimental formatter. See [#797](https://github.com/redhat-developer/vscode-xml/issues/797).
+ * Format CDATA sections without adding new lines. See [#585](https://github.com/redhat-developer/vscode-xml/issues/585).
+ * Generate grammar fails on first attempt with experimental formatter. See [lemminx#1382](https://github.com/eclipse/lemminx/issues/1382).
+ * Add tests for `xml.format.grammarAwareFormatting` setting and restore mixed content behaviour. See [lemminx#1364](https://github.com/eclipse/lemminx/pull/1364).
+ * Prevent exception in `DomElementFormatter.formatEndTagElement()`. See [lemminx#1361](https://github.com/eclipse/lemminx/issues/1361).
+ * NPE in CodeLens with empty XML file. See [lemminx#1396](https://github.com/eclipse/lemminx/issues/1396).
+ * Autoclose tag generates an unexpected `>`. See [lemminx#1221](https://github.com/eclipse/lemminx/issues/1221).
+
+### Build
+
+ * Fix issues reported by npm-audit. See [#799](https://github.com/redhat-developer/vscode-xml/pull/799).
+ * Bump decode-uri-component from 0.2.0 to 0.2.2. See [#828](https://github.com/redhat-developer/vscode-xml/pull/828).
+ * Bump yargs-parser from 5.0.0-security.0 to 5.0.1. See [#775](https://github.com/redhat-developer/vscode-xml/pull/775).
```

Comment:
If these reported due to security reasons, I would list their commits on the line above under "See ..."